app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesFragment.kt

```diff
@@ -195,8 +236,13 @@ class GradesFragment : FragmentForAccessingTumOnline<ExamList>(
     private fun calculateGradeDistribution(exams: List<Exam>): ArrayMap<String, Int> {
         val gradeDistribution = ArrayMap<String, Int>()
         exams.forEach { exam ->
-            val count = gradeDistribution[exam.grade] ?: 0
-            gradeDistribution[exam.grade] = count + 1
+            // The grade distribution now takes grades with more than one decimal place into account as well
+            var cleanGrade = exam.grade!!
+            if (cleanGrade.contains(Regex("[1-4],[0-9][0-9]+"))) {
```

Comment:
Same as above. Please avoid constructing the Regex object in the loop.

Reply:
:+1: